(BEDS-1371) API: pectra withdrawals endpoint adjustments

remoterami opened the pull request:
Based on #1419

remoterami committed:
feat: example consolidation data returned

remoterami committed:
refactor!: adjust to new api structure, pectra changes

remoterami committed:
feat(api): added CL reject reasons

remoterami committed:
refactor(api): unified param validation

LuccaBitfly commented on backend/pkg/api/types/validator_dashboard.go:
**suggestion:** You could add more faker fields which the frontend team would appreciate. See https://github.com/gobitfly/beaconchain/pull/1414#discussion_r2009717438

remoterami replied:
Okay I'll do that once approved in the linked PR, then I can rebase this branch and make use of the approach

remoterami replied:
Done, ptal

LuccaBitfly commented on frontend/types/api/validator_dashboard.ts:
```diff
@@ -233,20 +233,46 @@ export type GetValidatorDashboardTotalConsensusDepositsResponse = ApiDataRespons
-export interface VDBWithdrawalsTableRow {
-  epoch: number /* uint64 */;
-  slot: number /* uint64 */;
+export interface VDBWithdrawalsElTableRow {
```
**question:** This type seemingly has more fields than I can find in the design. Where do these come from?

- `executer`
- `is_missing_estimate`

remoterami replied:
Correct, executer is not present yet but should prob be added. I think it's analog to the deposits table where we split the EL transaction into `From` and `Depositor`. Will further discuss this
`is_missing_estimate` shouldn't be there for the EL withdrawals endpoint, I'll remove it.
For CL it was present in the past already, so I kept it. After looking into it it seems like it was not displayed anywhere directly but used as a logic entity instead. By convention the first row (on the first page) would always be the static "estimation" row, and this flag would indicate if it's not available for some reason. Doesn't feel too clean.. Maybe this should also be changed :sweat_smile:

remoterami replied:
`Executer` was renamed to `Withdrawer` & tx sender was (temporarily) removed, see design discussions for details. ptal

LuccaBitfly commented on frontend/types/api/validator_dashboard.ts:
```diff
@@ -233,20 +233,46 @@ export type GetValidatorDashboardTotalConsensusDepositsResponse = ApiDataRespons
+  tx_hash: Hash;
+  group_id: number /* uint64 */;
+  sender: Address;
+  executer: Address;
```
**typo:** executer -> executor?

LuccaBitfly commented on frontend/types/api/validator_dashboard.ts:
```diff
@@ -233,20 +233,46 @@ export type GetValidatorDashboardTotalConsensusDepositsResponse = ApiDataRespons
+  fee: string /* decimal.Decimal */;
+  is_missing_estimate: boolean;
+}
+export type GetValidatorDashboardExecutionLayerWithdrawalsResponse = ApiPagingResponse<VDBWithdrawalsElTableRow>;
```
**nitpick:** Don't know if it's applicable in this case, but frontend would appreciate a union type when there are mutually exclusive fields. See https://github.com/gobitfly/beaconchain/pull/1414#discussion_r2009730547.

LuccaBitfly replied:
won't do

LuccaBitfly commented on backend/pkg/api/handlers/input_validation.go:
```diff
@@ -373,19 +373,19 @@ func (v *validationError) parseSortOrder(order string) bool {
 }
 
-func checkSort[T enums.EnumFactory[T]](v *validationError, sortString string) *types.Sort[T] {
+func checkSort[T enums.EnumFactory[T]](v *validationError, sortString string) types.Sort[T] {
```
**praise:** good change (feel free to resolve)

remoterami committed:
feat: added basic consolidations endpoint

remoterami committed:
feat: split consolidations into el/cl, req+resp data

remoterami committed:
feat(api): return tailored mocked data

remoterami committed:
refactor(api)!: endpoint adjustments

remoterami committed:
feat(api): return tailored mocked data

enzo-bitfly committed:
refactor: rename <BcTable> prop to `is-loading`

enzo-bitfly committed:
feat: adapt `EL deposits table` for Pectra

Move fetching logic to `dashboard` root page component
See: BEDS-1362

enzo-bitfly committed:
refactor(variables.scss): replace spacing `px` values by `rem`

enzo-bitfly committed:
refactor(useCustomFetch): allow endpoint-specific `is_mocked` param to override global setting

LuccaBitfly reviewed (approved):
lgtm

enzo-bitfly committed:
feat: add `BcIconFlag`

enzo-bitfly committed:
feat: adapt `CL deposits table` for Pectra

-Move fetching logic to `dashboard` root page component
-Ensure immediate fetch for active tab on first page load

See: BEDS-1362

remoterami committed:
feat: added rejection reason mapping

enzo-bitfly committed:
feat: add `consolidations tab` with `EL consolidations table`

See: BEDS-1362

enzo-bitfly committed:
feat: add `CL consolidations table`

See: BEDS-1351

remoterami commented:
merged via #1480
